providers/PWAProvider.tsx

```
@@ -0,0 +1,233 @@
+}) => {
+  const [showMainContent, setShowMainContent] = useState(false);
+  const [lastActiveTime, setLastActiveTime] = useState(Date.now());
+  const router = useRouter();
+
+  useEffect(() => {
+    // Register service worker (enable in both dev and production for PWA testing)
+    if ('serviceWorker' in navigator) {
+      navigator.serviceWorker
+        .register('/sw.js', {
+          scope: '/',
+          updateViaCache: 'none' // Disable cache for service worker updates
```

ESLint

[error] Insert `,`

(prettier/prettier)